x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java

```diff
@@ -187,6 +202,15 @@ public void onFailure(Exception e, boolean receivedData) {
                 }
                 onAfter(List.of());
             }
+
+            @Override
+            public void onSkip(boolean skipRemaining) {
+                DataNodeRequestSender.this.skippedShards.incrementAndGet();
+                if (skipRemaining) {
+                    DataNodeRequestSender.this.skipRemaining = true;
+                }
+                onAfter(List.of());
```

Comment:
I think we should clear the shard failures for shards that are skipped; otherwise, we will still report failures.